### PR TITLE
Conversion of metadata dates and datetimes

### LINKED ISSUE #68: Convert mnp vars containing dates to Date format
**Is your feature request related to a problem? Please describe.**
"mnp" variables containing dates are not converted to Date format. Examples of such variables: mnpvispdt, mnpvisfdt, oldmnpvispdt, newmnpvispdt, mnpaedate, mnpvisstartdate, etc. 

**Describe the solution you'd like**
mnp variables containing dates should be converted to a Date variable with a .date ending, similar to conversion taking place for user-entered data.

**Describe alternatives you've considered**
/

**Additional context**
See secuTrial Export Formats Manual for complete list of mnp variables containing dates.

### PR DESCRIPTION
**Pull request by markomi**
Metadata variables containing dates or datetimes are now correctly recognized and converted to "Date" or "POSIXct" format.

In order to ensure that all metadata variables containing dates or datetimes are converted, I went throught the "secuTrial® Export Formats" Manual and made an exhaustive list of all variables that need to be converted.

I'd appreciate if you would test the code with your datasets. Comments are very welcome.

This PR resolves #68 .

### CODE REVIEW

File: R/dates_secuTrial.R [hunk 1]
```diff
@@ -27,58 +27,19 @@
...
 dates_secuTrial.secuTrialdata <- function(object, ...) {
   if (object$export_options$dated) warning("dates already added")
-
   table_names <- object$export_options$data_names
   names(table_names) <- NULL
-  # not for meta tables
-  table_names <- table_names[!table_names %in% object$export_options$meta_names]
   # get language and internationalization dictionary for items table
   dict <- object$export_options$dict_items
+  obs <- lapply(table_names, function(obj) .convert_dates(obj, object, dict, ...))
```
Comment (aghaynes) [inline R/dates_secuTrial.R]: can this be simplified to 
```
obs <- lapply(table_names, .convert_dates, object = object, dict = dict, ...)
```
?

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# secuTrialR 0.6.2
```
Comment (PatrickRWright) [inline NEWS.md]: This should be 0.6.3

**Comment by PatrickRWright** [comment]
I think you forgot to knit README.Rmd

File: R/dates_secuTrial.R (hunk 1 above)
Comment (markomi) [reply R/dates_secuTrial.R]: you're right. i'll implement the change. thanks!

File: NEWS.md (hunk 2 above)
Comment (markomi) [reply NEWS.md]: well spotted. thanks

File: R/dates_secuTrial.R [hunk 3]
```diff
@@ -97,33 +58,29 @@ dates_secuTrial.data.frame <- function(data, datevars, timevars, dateformat, dat
       newdatecol <- dates_secuTrial(data[, x], dateformat)
       # check for conversion of all else warn
       if (length(which(is.na(newdatecol))) > length(which(is.na(data[, x])))) {
-        warning(paste0("Not all dates were converted for\n",
+        if (warn) warning(paste0("Not all dates were converted for\n",
```
Comment (PatrickRWright) [inline R/dates_secuTrial.R]: As we just discussed I would prefer this not being disableable.Same goes for line 76.

Comment (aghaynes) [reply R/dates_secuTrial.R]: I would agree with that...if the user wants to silence warnings they can do so using the normal R method (i.e. via options)

File: README.md [hunk 4]
```diff
@@ -451,11 +451,11 @@ return_random_cases(ctu05, percent = 0.25, seed = 1337, date = "2019-03-18",
...
-## 1  RPACK-INS-011 Inselspital Bern (RPACK)      2019-04-11
-## 4  RPACK-INS-014 Inselspital Bern (RPACK)      2019-04-14
-## 41 RPACK-CBE-004   Charité Berlin (RPACK)      2019-04-04
-## 3  RPACK-CBE-003   Charité Berlin (RPACK)      2019-04-03
+##          mnpaid                   centre mnpvisstartdate
+## 2 RPACK-INS-012 Inselspital Bern (RPACK)      2019-04-12
+## 4 RPACK-INS-014 Inselspital Bern (RPACK)      2019-04-14
+## 5 RPACK-CBE-005   Charité Berlin (RPACK)      2019-04-05
+## 3 RPACK-CBE-003   Charité Berlin (RPACK)      2019-04-03
 ```
```
Comment (PatrickRWright) [inline README.md]: This is not related directly to your PR but its confusing to me. This should not change.

Any ideas? @markomi @aghaynes

Comment (aghaynes) [reply README.md]: different systems (e.g. unix/Windows) use different random number tables I think.. dont know for sure though...
We can check...
On windows...
```
> set.seed(1234)
> rnorm(10)
 [1] -1.2070657  0.2774292  1.0844412 -2.3456977  0.4291247  0.5060559 -0.5747400 -0.5466319 -0.5644520
[10] -0.8900378
```
Can you both run those couple of lines too?

Comment (PatrickRWright) [reply README.md]:
```
> set.seed(1234)
> rnorm(10)
 [1] -1.2070657  0.2774292  1.0844412 -2.3456977  0.4291247  0.5060559 -0.5747400
 [8] -0.5466319 -0.5644520 -0.8900378

Comment (markomi) [reply README.md]: platform       x86_64-pc-linux-gnu 
version.string R version 3.6.1 (2019-07-05)
```
> set.seed(1234)
> rnorm(10)
[1] -1.2070657  0.2774292  1.0844412 -2.3456977  0.4291247  0.5060559
 [7] -0.5747400 -0.5466319 -0.5644520 -0.8900378
```

Comment (PatrickRWright) [reply README.md]: So this is likely due to openSUSE since all results that we posted here are the same and Milica was using an openSUSE OS yesterday.

**Comment by PatrickRWright** [comment]
I ran some manual checks on local datasets and it looks fine to me. 

Thank you @markomi

File: R/internals.R [hunk 5]
```diff
@@ -173,3 +173,75 @@ convertnames <- function(df, format){
...
+  }
+  if (warn & length(meta_datevars) == 0){
+    warning(paste("no dates detected in metadata of ", obj))
+  }
+  if (warn & length(timevars) == 0) {
+    warning(paste("no datetimes detected in ", obj))
+  }
+  if (warn & length(meta_timevars) == 0) {
+    warning(paste("no datetimes detected in metadata of ", obj))
+  }
```
Comment (aghaynes) [inline R/internals.R]: In line with other comments... can you remove the warn from the warn condition?

File: R/internals.R [hunk 6]
```diff
@@ -173,3 +173,75 @@ convertnames <- function(df, format){
...
+
+# Converts dates using dates_secuTrial dispatcher
+#
+# @param object - secuTrial object containing imported secuTrial data
+# @param obj - string containing the name of a data frame within 'object'
+# @param dict - data frame containing language and internationalization dictionary for items table
+# @param warn - logical which determines whether warnings will be shown (TRUE) or not (FALSE)
+# @return data frame containing object[[obj]] with converted dates and datetimes, if applicable
+#
+.convert_dates <- function(obj, object, dict, warn = FALSE, ...) {
```
Comment (aghaynes) [inline R/internals.R]: the warn option is probably not needed

File: R/internals.R (hunk 5 above)
Comment (markomi) [reply R/internals.R]: so i think in this case we used to have the `if(warn` even prior to this PR. i've just added it for meta_datevars and meta_timevars similarly to what we had for timevars and datevars before.

could you check the previous implementation of R/dates_secuTrial.R on lines 109, 126 (marked red above) and let me know what you think? thanks!

Comment (PatrickRWright) [reply R/internals.R]: For now I would suggest to keep the `warn` here because I don't think we want these specific warnings to be returned as a default.

Comment (aghaynes) [reply R/internals.R]: ok, then we leave it

Comment (markomi) [reply R/internals.R]: thanks for having another look at this.

**Comment by PatrickRWright** [comment]
Can I merge?

**Comment by aghaynes** [comment]
Fine with me

**Pull request merged**